pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -715,9 +715,11 @@ const storeKafkaTemplate = `{{- define "storeKafka" -}}
 brokers {{.Brokers}}
 default_topic {{.Topic}}
 {{ if .Target.Secret -}}
+{{ $tlsCert := .SecretPath "tls.crt" }}
+{{ $tlsKey := .SecretPath "tls.key" }}
 ssl_ca_cert '{{ .SecretPath "ca-bundle.crt"}}'
-ssl_client_cert '{{ .SecretPath "tls.crt"}}'
-ssl_client_cert_key '{{ .SecretPath "tls.key"}}'
+ssl_client_cert "#{File.exist?('{{ $tlsCert }}') ? '{{ $tlsCert }}' : use_nil}"
```

Comment:
Isn't there code elsewhere where we check for the secret and its keys prior to using the template which means we remove one more conditional from configuration at run time.  This should be our goal: make the configuration "what you see is what you get" as much as is possible

Reply:
We actually don't check the secret or its contents when generating TLS configuration for our outputs. We enable the TLS config block if the forwarder spec has a sercetRef. The normalizer is only checking if the sercetRef exists as a sercret resource nothing more.